Update CleanABAP.md - new rule "use snake_case" (#320)
SAP/styleguides: clean-abap/CleanABAP.md

> Read more in _Chapter 2: Meaningful Names: Use Pronounceable Names_ of [Robert C. Martin's _Clean Code_]

### Use snake_case
> [Clean ABAP](#clean-abap) > [Content](#content) > [Names](#names) > [This section](#use-snake_case)

> ABAP is case insensitive and limits the number of available letters for naming all kind of DDIC-objects
> and variables. For example the maximum length of a method name is 30 characters. The defacto naming convention
> in ABAP is `snake_case`, because ABAP is not case sensitive and camelCase is overwritten when the Pretty Printer / Formatter is activated and not all team members use the 'Keep Camel Case Identifiers' setting in the ADT ABAP Formatter. Also `camelCase` doesn't
> affect DDIC-objects.
> We recommend using `snake_case` consistently instead of `flatcase` or `UPPERCASE`. When you reach the maximum 
> length of an object, try to conscientiously use abbreviations (see [Use same abbreviations everywhere](#use-same-abbreviations-everywhere)).

Reviewer: Thanks for the change, but now the whole section reads kind of awkwardly because of the repetition. Here's my suggestion on a more concise explanation:

```suggestion
ABAP is case insensitive which is why we recommend following the convention to use `snake_case` consistently.

There's a character limit for names, e.g. 30 characters for methods. When you reach the maximum length of an object, don't fall back to using `flatcase` or `UPPERCASE`. Try to conscientiously use abbreviations instead (see [Use same abbreviations everywhere](#use-same-abbreviations-everywhere)).
```

Reviewer: I left out the sentence on the Formatter because I don't think this argument holds up. The whole team should use the same formatter settings, see rule "Use your Pretty Printer team settings".

```ABAP

> " a variable which contains the reponse time measured in milliseconds

Reviewer: ```suggestion
" a variable which contains the response time measured in milliseconds
```

response_time_in_millisec  " ok

> responsetimeinmilliseconds " bad practice

Reviewer: Can you think of a better example? `response_time_in_milliseconds` still only has 29 characters, so it doesn't really get the point across.

Author: maybe like this?
max_response_time_in_millisec
maxresponsetimeinmilliseconds

Reviewer: LGTM 👍

> Read more in _Chapter 2: Meaningful Names: Use Pronounceable Names_ of [Robert C. Martin's _Clean Code_]

### Use snake_case

> > [Clean ABAP](#clean-abap) > [Content](#content) > [Names](#names) > [This section](#use-snake-case)

Reviewer 2: ```suggestion
> [Clean ABAP](#clean-abap) > [Content](#content) > [Names](#names) > [This section](#use-snake_case)
```

ABAP is case insensitive and limits the number of available letters for naming all kind of DDIC-objects

> and variables. For exampe the maximum length of a method name is 30 characters. The defacto naming convention

Reviewer: Spelling error: 

```suggestion
and variables. For example the maximum length of a method name is 30 characters. The defacto naming convention
```

in ABAP is `snake_case`, because camelCase is overwritten when the Pretty Printer is activated and not all team

> members use the 'Keep Camel Case Identifiers' setting in the ADT ABAP Formatter. Also `camelCase` doesn't

Reviewer: I think we should reformulate the reason given. 

The main reason for `snake_case` to be the main convention is the fact that ABAP is case insensitive and the system will disregard the case of the name in several places. This should be presented as the main argument. The fact about the pretty printer is a nice addition, but shouldn't be presented as the main argument in my opinion.

Author: good one.

affect DDIC-objects.
We recommend using `snake_case` consistently instead of `flatcase` or `UPPERCASE`. When you reach the maximum 

> length of an object, try to conscientiously use abbreviations (also see [Avoid abbreviations](#avoid-abbreviations)).

Reviewer: ```suggestion
length of an object, try to conscientiously use abbreviations (see [Use same abbreviations everywhere](#use-same-abbreviations-everywhere)).
```

Reviewer: Or are you not actually using "conscientiously" in the sense of always using the same abbreviations? If you mean something else, then maybe a different word would fit better.
